core/src/main/scala/com/mesosphere/usi/core/SchedulerFactory.scala

```diff
@@ -48,20 +51,26 @@ private[usi] trait MesosFlowFactory {
   * @param metrics Metrics
   */
 class SchedulerFactory private (
-    client: MesosClient,
+    clientSettings: MesosClientSettings,
+    frameworkInfo: FrameworkInfo,
+    authorization: Option[CredentialsProvider],
     podRecordRepository: PodRecordRepository,
     schedulerSettings: SchedulerSettings,
-    metrics: Metrics)(implicit ec: ExecutionContext)
+    metrics: Metrics)(implicit ec: ExecutionContext, system: ActorSystem, materializer: Materializer)
     extends SchedulerLogicFactory
     with PersistenceFlowFactory
     with SuppressReviveFactory
     with MesosFlowFactory
     with StrictLogging {
 
-  val frameworkInfo: FrameworkInfo = client.frameworkInfo
-
   override def newMesosFlow: Flow[MesosCall, MesosEvent, NotUsed] =
-    Flow.fromSinkAndSourceCoupled(logMesosCallException(client.mesosSink), client.mesosSource)
+    RestartFlow.withBackoff(5.seconds, 30.seconds, 0.2, 10) { () =>
+      // TODO: Inject client factory
+      val flow = MesosClient(clientSettings, frameworkInfo, authorization).map { client =>
```

Comment:
I tried to refactor the client so that it would not connect immediately but failed. I don't want to to touch the client much. Unfortunately we always assume that the client is available. See below.